packages/ag-charts-community/src/chart/series/seriesAreaFocusManager.ts

```diff
@@ -54,15 +54,21 @@ export class SeriesAreaFocusManager extends BaseManager {
     }
 
     public seriesChanged(series: Series<any, SeriesProperties<any>>[]) {
-        this.series = [...series].sort((a, b) => {
-            let fpA = a.properties.focusPriority ?? Infinity;
-            let fpB = b.properties.focusPriority ?? Infinity;
+        const unsortedSeries = series.map((series, index) => {
+            return { series, index };
+        });
+        const sortedSeries = unsortedSeries.sort((a, b) => {
+            let fpA = a.series.properties.focusPriority ?? Infinity;
+            let fpB = b.series.properties.focusPriority ?? Infinity;
             if (fpA === fpB) {
-                [fpA, fpB] = [a._declarationOrder, b._declarationOrder];
```

Comment:
Hold up - we need to respect declaration order, or things can get very messy after chart option updates which add/remove series or re-order them.

Reply:
In that case, would it not be better to throw an error if the `_declaractionOrders` are equal? (rather than returning 0)

Reply:
Let's roll with this and revert to `_declarationOrder` on `latest` post release.